### DISCUSSION TITLE
Multiple concrete numbering using same abstract numbering definition

### DISCUSSION THREAD

**Discussion by BrentFarese** [dolanmiu/docx#1136] [Q&A]
Aline has a document model for lists that is similar to Word.  You can have multiple concrete numbering reference the same abstract numbring definition.  You can also specify "level overrides" as in Word.

The [DOCX numbering documentation](https://docx.js.org/#/usage/numbering) seems to reference the old non-declarative way of creating numbering.

Is there a way to have multiple concrete numbering reference the same abstract definition?  The declarative approach doesn't seem to support it, unless I'm missing something.

Thanks!

**Comment by dolanmiu** [comment]
It is outdated documentation

Needs updating

The declarative way is done in a few examples in the demo folder

Check those for reference

**Reply by BrentFarese** [reply]
Thanks @dolanmiu!  We use the declarative way in our code.  But my question remains, is there a way in the declarative fashion to have multiple numbering instances to point to the same abstract numbering definition?  Or is that not supported (it seems the Word spec supports it).

Can you add properties to the numbering instance (e.g., `lvlOverride` from Word)?  Maybe these are things that are just not supported yet but wanted to check.

**Reply by dolanmiu** [reply]
AFAIK, it does do this already:

https://github.com/dolanmiu/docx/blob/master/src/file/numbering/numbering.ts#L193
https://github.com/dolanmiu/docx/blob/master/src/file/paragraph/properties.ts#L172-L180

Correct me if im wrong

**Reply by BrentFarese** [reply]
I think you are right that [prepForXML](https://github.com/dolanmiu/docx/blob/master/src/file/paragraph/properties.ts#L172-L180) as part of the Paragraph class will [createConcreteNumberingInstances](https://github.com/dolanmiu/docx/blob/eb5a4f96203496bbf43f02e3a066e5ffc9d2e409/src/file/numbering/numbering.ts#L193) that refer to the same abstract definition.   If 2 paragraphs have a different `numbering.reference` and both those numbering instances point to the same abstract definition, should work well.

Is there any way to add `lvlOverride` to those concrete numbering instances though?  It seems that maybe that part of my question is not supported?  I haven't seen anything on the Paragraph class or elsewhere that might let us specify `lvlOverrides` for a given concrete numbering instance.  We are trying to model our lists as close to Word as possible so we can export to DOCX perfectly (we are trying to support "Office editing" in the browser) and can't figure out how to use `lvlOverride` correctly for concrete numbering instances as part of the library.  

OOXML Spec Part I, section 17.9.8 has a bunch of properties in `lvlOverride` that we're trying to support.

Thanks for the great library!  If it's not supported, maybe we'll try a PR to support it.

**Reply by dolanmiu** [reply]
Does seems like the `lvlOverride` feature is there:

https://github.com/dolanmiu/docx/blob/master/src/file/numbering/concrete-numbering.spec.ts#L15-L17
https://github.com/dolanmiu/docx/blob/master/src/file/numbering/num.ts#L26-L29

**Reply by BrentFarese** [reply]
Yea it seems like `lvlOverride` is there, we just can't figure out how to access it declaratively.  See:

https://github.com/dolanmiu/docx/blob/eb5a4f96203496bbf43f02e3a066e5ffc9d2e409/src/file/core-properties/properties.ts#L23
https://github.com/dolanmiu/docx/blob/eb5a4f96203496bbf43f02e3a066e5ffc9d2e409/src/file/numbering/level.ts#L82

Maybe there is a non-declarative way to do it?  Also, `lvlOverride` is missing available props from the OOXML spec, which is alright but I suppose there is a feature gap.  We may just try and work around it in our modeling and not rely on `lvlOverride` but this discussion has been super helpful!

Thank you.
